model_analyzer/analyzer.py

```diff
@@ -114,7 +128,7 @@ def write_and_export_results(self):
             self._plot_manager.compile_and_export_plots()
 
             # Export individual model summaries
-            for model in self._config.model_names:
+            for model in self._config.model_names[:self._model_index]:
```

Comment:
Why does the top level analyzer need a model index? Shouldn't it be analyzing all models?

Reply:
Yes, but that keeps track of how many models were profiled, so that only the results for those models are queried in the report generation part.